legend-engine-xt-graphQL-query/src/test/java/org/finos/legend/engine/query/graphQL/api/test/TestGraphQLAPI.java

```diff
@@ -245,6 +330,82 @@ public void testGraphQLDebugGenerateGraphFetchDevAPI()
         Assert.assertEquals(expected, DEPRECATED_PureGrammarComposerCore.Builder.newInstance().withRenderStyle(RenderStyle.PRETTY).build().processGraphFetchTree(((GraphFetchResult) response.getEntity()).graphFetchTree, 2));
     }
 
+
+    @Test
+    public void testCacheUsed() throws Exception
+    {
+        ExecutionCache<GraphQLCacheKey, List<SerializedNamedPlans>> planCache = ExecutionCacheBuilder.buildExecutionCacheFromGuavaCache(CacheBuilder.newBuilder().recordStats().build());
```

Comment:
Should we update the Legend engine server to inject a cache ?

Reply:
Also .. how do we want to manage the cache  For e.g use Guava to say prune entries after a certain size etc ?